### PR TITLE
Modify `list` command

### PR DESCRIPTION
**Pull request by pwjj2000**
Outstanding issues **(completed)**:
- [x]  JUnit tests not written
- [x]  Javadoc headers for public methods
- [x]  Creating related exceptions
- [x]  Amending `patients` and `appts` lists to be side-by-side instead of top-and-bottom
- [x]  Appointment list unsorted
- [x]  CheckStyle errors

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Appointment.java [hunk 1]
```diff
@@ -99,10 +107,22 @@ public void unmark() {
         this.isMarked = false;
     }
 
+    public void setPatient(Person patient) {
+        this.patient = patient;
+    }
```
Comment (JovonLim) [inline src/main/java/seedu/address/model/person/Appointment.java:112]: Would it be sufficient if we kept a reference to the patient name instead? Then we can store it in the Json without needing to iterate thru all appointments to set the patient during conversion.

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ListCommand object.
+ */
+public class ListCommandParser implements Parser<ListCommand> {
```
Comment (JovonLim) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java:11]: Can use a matcher instead to check for the patients or appts words in the user input

**Review by JovonLim — APPROVED** [review]
LGTM overall with some potential improvements

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 3]
```diff
@@ -12,13 +13,36 @@ public class ListCommand extends Command {
...
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (this.type.equals("patients")) {
+            requireNonNull(model);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(MESSAGE_SUCCESS_PATIENTS);
+        } else {
+            requireNonNull(model);
+            model.updateFilteredAppointmentList(PREDICATE_SHOW_ALL_APPOINTMENTS);
+            return new CommandResult(MESSAGE_SUCCESS_APPOINTMENTS);
+        }
```
Comment (NicholasTYD) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:39]:
```suggestion
        requireNonNull(model);
        if (this.type.equals("patients")) {
            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
            return new CommandResult(MESSAGE_SUCCESS_PATIENTS);
        } else {
            model.updateFilteredAppointmentList(PREDICATE_SHOW_ALL_APPOINTMENTS);
            return new CommandResult(MESSAGE_SUCCESS_APPOINTMENTS);
        }
```
Perhaps the requireNonNull(model) line could be removed and placed at the top before the if else block? This way we can reduce code duplication.

File: src/main/java/seedu/address/model/person/Appointment.java [hunk 4]
```diff
@@ -122,8 +142,20 @@ public boolean equals(Object other) {
         }
 
         Appointment otherAppointment = (Appointment) other;
-        return otherAppointment.reason.equals(reason)
+
+        return otherAppointment.patient.getName().equals(patient.getName())
```
Comment (NicholasTYD) [inline src/main/java/seedu/address/model/person/Appointment.java:146]:
```suggestion
        return otherAppointment.patient.equals(patient)
```
Is it possible to shorten this line to code to this? Would it break anything?

File: src/main/java/seedu/address/model/person/Appointment.java (hunk 1 above)
Comment (NicholasTYD) [reply src/main/java/seedu/address/model/person/Appointment.java:112]: Could it work if modify the constructor to take in the appointment's owner, so we don't have to manually set it all the time?

**Review by NicholasTYD — APPROVED** [review]
LGTM. There's a few really minor nitpicks that shouldn't matter too much, but overall nice job!

**Pull request merged**